subcommand/controller/command.go

```diff
@@ -0,0 +1,117 @@
+package controller
+
+import (
+	"context"
+	"flag"
+	"os"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/hashicorp/consul-ecs/awsutil"
+	"github.com/hashicorp/consul-ecs/controller"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+)
+
+const (
+	flagAgentSecretARN   = "agent-secret-arn"
+	flagSecretNamePrefix = "secret-name-prefix"
+
+	consulCACertEnvVar = "CONSUL_CACERT_PEM"
+)
+
+type Command struct {
+	UI                   cli.Ui
+	flagAgentSecretARN   string
+	flagSecretNamePrefix string
+
+	log     hclog.Logger
+	flagSet *flag.FlagSet
+	once    sync.Once
+}
+
+func (c *Command) init() {
+	c.flagSet = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flagSet.StringVar(&c.flagAgentSecretARN, flagAgentSecretARN, "", "ARN of AWS Secrets Manager secret")
+	c.flagSet.StringVar(&c.flagSecretNamePrefix, flagSecretNamePrefix, "", "The prefix for secret names stored in AWS Secrets Manager")
+
+	c.log = hclog.New(nil)
+}
+
+func (c *Command) Run(args []string) int {
+	c.once.Do(c.init)
+	if err := c.flagSet.Parse(args); err != nil {
+		return 1
+	}
+
+	err := c.run()
+	if err != nil {
+		c.log.Error(err.Error())
```

Comment:
🙏 banish `c.UI` 😂

Reply:
That's from your POC I think 😄 and I completely agree!